### PR TITLE
arc: enable thread local storage

### PR DESCRIPTION
**Pull request by dcpleung**
This adds the necessary bits to support thread local storage
in ARC.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

### CODE REVIEW

File: include/arch/arc/v2/linker.ld [hunk 1]
```diff
@@ -220,5 +221,11 @@ SECTIONS {
 		KEEP(*(.gnu.attributes))
 	}
 
-	/DISCARD/ : { *(.note.GNU-stack) }
+	/DISCARD/ : {
```
Comment (abrodkin) [inline include/arch/arc/v2/linker.ld:224]: Even though that clean-up shouldn't hurt, but does it have anything to do with TLS really?

**Review by abrodkin — APPROVED** [review]
Cool, thanks for taking care of ARC here! One minor question on linker sections clean-up.

File: include/arch/arc/v2/linker.ld (hunk 1 above)
Comment (dcpleung) [reply include/arch/arc/v2/linker.ld:224]: The `.got*` sections are generated when there are `__thread` variables. So they need to be discarded.

Comment (abrodkin) [reply include/arch/arc/v2/linker.ld:224]: Makes sense then, didn't think about it.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by ruuddw** [comment]
Thanks @dcpleung - have you (or someone else) run some tests on nSIM or ARC boards for this, or only building?
If not, should we run some tests before merging?

**Comment by dcpleung** [comment]
> Thanks @dcpleung - have you (or someone else) run some tests on nSIM or ARC boards for this, or only building?
> If not, should we run some tests before merging?

I ran nSIM with tests under `tests/kernel` and this passed all tests, except `tests/kernel/fatal/exception/kernel.common.stack_protection` (but it also fails on master). No new failed kernel tests were introduced.

**Pull request merged**